pgmpy/inference/continuous/base.py

```diff
@@ -0,0 +1,365 @@
+    ...         log_pdf = 0.5 * np.float(np.dot(sub_vec.T, grad))
+    ...         return grad, log_pdf
+    >>> mean = np.array([[1], [1]])
+    >>> covariance = np.array([[1, 0.2], [0.2, 7]])
+    >>> model = JointGaussianDistribution(mean, covariance)
+    >>> dist_param = np.array([[0.1], [0.9]])
+    >>> grad_logp, logp = GradLogGaussian(dist_param, model).get_gradient_log_pdf()
+    >>> logp
+    -0.4054597701149426
+    >>> grad_logp
+    array([[ 0.90229885],
+           [-0.01149425]])
+    """
+
+    def __init__(self, distribution_param, model):
+
+        if isinstance(distribution_param, (np.matrix, np.ndarray, list, tuple, set, frozenset)):
+            distribution_param = np.array(distribution_param).flatten()
+            distribution_param = np.reshape(distribution_param, (len(distribution_param), 1))
+        else:
+            raise TypeError("distribution_param should be a 1d array type object")
+
+        self.distribution_param = distribution_param
+        self.model = model
+        # The gradient of probability distribution at position
+        self.grad_log = None
+        # The gradient log of probability distribution at position
+        self.log_pdf = None
+
+    def get_gradient_log_pdf(self):
+        """
+        Returns the gradient log and log of model at given position
+
+        Returns
+        -------
+        numpy.array: Representing gradient log of model at given position
+
+        float: Representing log of model at given position
+
+        Example
+        --------
+        >>> # Using implementation of GradLogPDFGaussian
+        >>> from pgmpy.inference.continuous import GradLogPDFGaussian
+        >>> from pgmpy.models import JointGaussianDistribution
+        >>> import numpy as np
+        >>> mean = np.array([[1], [1]])
+        >>> covariance = np.array([[1, 0.2], [0.2, 7]])
+        >>> model = JointGaussianDistribution(mean, covariance)
+        >>> dist_param = np.array([[0.1], [0.9]])
+        >>> grad_logp, logp = GradLogPDFGaussian(dist_param, model).get_gradient_log_pdf()
+        >>> logp
+        -0.4054597701149426
+        >>> grad_logp
+        array([[ 0.90229885],
+               [-0.01149425]])
+        """
+        return self.grad_log, self.log_pdf
+
+
+class GradLogPDFGaussian(BaseGradLogPDF):
+    """
+    Class for finding gradient and gradient log of Joint Gaussian Distribution
+    Inherits pgmpy.inference.base_continuous.BaseGradLogPDF
+    Here model must be an instance of JointGaussianDistribution
+
+    Example
+    -------
+    >>> from pgmpy.inference.continuous import GradLogPDFGaussian
+    >>> from pgmpy.models import JointGaussianDistribution
+    >>> import numpy as np
+    >>> mean = np.array([[1], [1]])
+    >>> covariance = np.array([[1, 0.2], [0.2, 7]])
+    >>> model = JointGaussianDistribution(mean, covariance)
+    >>> dist_param = np.array([[0.1], [0.9]])
+    >>> grad_logp, logp = GradLogPDFGaussian(dist_param, model).get_gradient_log_pdf()
```

Comment:
@khalibartan In the docstring you have mentioned that `dist_param` should be a 1-D array but here you are accepting a 2-D array.